scheduler/views.py

```diff
@@ -18,16 +18,25 @@ def post(self, request, *args, **kwargs):
             status = 400
             message = {"contacts.0.wa_id": ["This field is required."]}
             return Response(message, status=status)
-        
+
         content = ReminderContent.objects.last()
 
         # Default to 23 hours but allow us to overwrite it
         delay = int(request.GET.get('hour_delay', 23))
         scheduled_for = timezone.now() + timedelta(hours=delay)
-        ReminderSchedule.objects.create(
+        new_reminder = ReminderSchedule.objects.create(
             schedule_time=scheduled_for,
             recipient_id=recipient_id,
             content=content
         )
 
+        existing_reminders = ReminderSchedule.objects.filter(
+            recipient_id=recipient_id,
+            sent_time__isnull=True,
+            cancelled=False).exclude(pk=new_reminder.pk)
+
+        for reminder in existing_reminders:
+            reminder.cancelled = True
+            reminder.save()
```

Comment:
Could we not rather do a `.filter(...).update(cancelled=True)` to do them all at once?

I'm happy to leave this without an index for now, hopefully the index on `sent_time` will be enough for now, and if we find that this endpoint is getting slow, then we can always add an index on `recipient_id`

Reply:
I've decided to leave `recipient_id` unindexed. I feel like it's easier to identify if we need the index and add it than it will be to figure out if the index is causing problems.